android/src/main/java/com/swmansion/rnscreens/ScreenStackHeaderConfig.kt

```diff
@@ -149,26 +156,36 @@ class ScreenStackHeaderConfig(context: Context) : ViewGroup(context) {
         }
 
         if (isHeaderHidden) {
-            if (toolbar.parent != null) {
-                screenFragment?.removeToolbar()
-            }
+            screenFragment?.removeToolbar()
             return
         }
 
+        // Reattach the header, when the developer tries to change the header type
+        if (screenStackHeader?.loadedHeaderType != screen?.headerType) {
+            screenStackHeader?.updateHeaderType()
+            screenFragment?.removeToolbar()
```

Comment:
Why do we remove it here?

Reply:
That's because of updating the `layoutParams` of appBarLayout. Basically, when we're changing the type of appBar, its layout is being changed, resulting in wrong position of the views. For that reason we're removing all of its views (they will be added further in onUpdate) and setting references of toolbar and collapsingtoolbarlayout to null to ensure that we won't use old toolbars

Reply:
Resolved in https://github.com/software-mansion/react-native-screens/pull/1874/commits/4c5612a3a6acc7788da2585a399970ace5dd6124.